bitnami/common/templates/_labels.tpl

```diff
@@ -4,20 +4,53 @@ SPDX-License-Identifier: APACHE-2.0
 */}}
 
 {{/* vim: set filetype=mustache: */}}
+
 {{/*
 Kubernetes standard labels
+{{ include "common.labels.standard" (dict "customLabels" .Values.commonLabels "context" $) -}}
 */}}
 {{- define "common.labels.standard" -}}
+{{- if and (hasKey . "customLabels") (hasKey . "context") -}}
+{{ merge
+        (include "common.tplvalues.render" (dict "value" .customLabels "context" .context) | fromYaml)
+        (dict
+            "app.kubernetes.io/name" (include "common.names.name" .context)
+            "helm.sh/chart" (include "common.names.chart" .context)
+            "app.kubernetes.io/instance" .context.Release.Name
+            "app.kubernetes.io/managed-by" .context.Release.Service
+        )
+    | toYaml
+}}
+{{- else -}}
 app.kubernetes.io/name: {{ include "common.names.name" . }}
 helm.sh/chart: {{ include "common.names.chart" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
 {{- end -}}
+{{- end -}}
 
 {{/*
-Labels to use on deploy.spec.selector.matchLabels and svc.spec.selector
+Labels used on immutable fields such as deploy.spec.selector.matchLabels or svc.spec.selector
+{{ include "common.labels.matchLabels" (dict "customLabels" .Values.podLabels "context" $) -}}
+
+We don't want to loop over custom labels appending them to the selector
+since it's very likely that it will break deployments, services, etc.
+However, it's important to overwrite the standard labels if the user
+overwrote them on metadata.labels fields.
 */}}
 {{- define "common.labels.matchLabels" -}}
+{{- if and (hasKey . "customLabels") (hasKey . "context") -}}
+{{- $customLabels := pick .customLabels "app.kubernetes.io/name" "app.kubernetes.io/instance" -}}
```

Comment:
won't work if customLabels is a string.

you should do the pick after the merge :)

Reply:
Done at https://github.com/bitnami/charts/pull/18154/commits/39894f99c3c6c31ba8c6b5e4411f61ce0ee6df87